drivers/soundwire/intel_init.c

```diff
@@ -363,6 +364,47 @@ void sdw_intel_exit(struct sdw_intel_ctx *ctx)
 }
 EXPORT_SYMBOL_NS(sdw_intel_exit, SOUNDWIRE_INTEL_INIT);
 
+static int intel_disable_runtime_pm_child_device(struct device *dev, void *data)
+{
+	struct sdw_slave *slave = dev_to_sdw_dev(dev);
+	unsigned long time;
+
+	if (!slave->probed) {
+		dev_dbg(dev, "skipping device, no probed driver\n");
+		return 0;
+	}
+	if (!slave->dev_num_sticky) {
+		dev_dbg(dev, "skipping device, never detected on bus\n");
+		return 0;
+	}
+
+	pm_runtime_disable(dev);
```

Comment:
I think this is conceptually broken.

pm_runtime_disable() only means that the device can no longer enter pm_runtime suspend. It doesn't do what is suggested in the commit message "wait for peripherals to being initialized"

In addition, if something else tries to do a pm_runtime_get_sync() or pm_runtime_resume(), an error code -EACCESS will be thrown.

We should only use pm_runtime_disable() in the .remove functions or AFTER resuming the devices fully.